pkg/services/live/telemetry/telegraf/convert.go

```diff
@@ -0,0 +1,325 @@
+			err := frame.append(m)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			frameKeyOrder = append(frameKeyOrder, frameKey)
+			frame = newMetricFrameLabelsColumn(m, c.useFloat64Numbers)
+			err := frame.append(m)
+			if err != nil {
+				return nil, err
+			}
+			metricFrames[frameKey] = frame
+		}
+	}
+
+	frameWrappers := make([]telemetry.FrameWrapper, 0, len(metricFrames))
+	for _, key := range frameKeyOrder {
+		frameWrappers = append(frameWrappers, metricFrames[key])
+	}
+
+	return frameWrappers, nil
+}
+
+type metricFrame struct {
+	useFloatNumbers bool
+	key             string
+	fields          []*data.Field
+	fieldCache      map[string]int
+}
+
+// newMetricFrame will return a new frame with length 1.
+func newMetricFrame(m influx.Metric, useFloatNumbers bool) *metricFrame {
+	s := &metricFrame{
+		useFloatNumbers: useFloatNumbers,
+		key:             m.Name(),
+		fields:          make([]*data.Field, 1),
+	}
+	s.fields[0] = data.NewField("time", nil, []time.Time{m.Time()})
+	return s
+}
+
+// newMetricFrame will return a new frame with length 1.
+func newMetricFrameLabelsColumn(m influx.Metric, useFloatNumbers bool) *metricFrame {
+	s := &metricFrame{
+		useFloatNumbers: useFloatNumbers,
+		key:             m.Name(),
+		fields:          make([]*data.Field, 2),
+		fieldCache:      map[string]int{},
+	}
+	s.fields[0] = data.NewField("labels", nil, []string{})
+	s.fields[1] = data.NewField("time", nil, []time.Time{})
+	return s
+}
+
+// Key returns a key which describes Frame metrics.
+func (s *metricFrame) Key() string {
+	return s.key
+}
+
+// Frame transforms metricFrame to Grafana data.Frame.
+func (s *metricFrame) Frame() *data.Frame {
+	return data.NewFrame(s.key, s.fields...)
+}
+
+// extend existing metricFrame fields.
+func (s *metricFrame) extend(m influx.Metric) error {
+	fields := m.FieldList()
+	sort.Slice(fields, func(i, j int) bool {
+		return fields[i].Key < fields[j].Key
+	})
+	labels := tagsToLabels(m.TagList())
+	for _, f := range fields {
+		ft, v, err := s.getFieldTypeAndValue(f)
+		if err != nil {
+			return err
+		}
+		field := data.NewFieldFromFieldType(ft, 1)
+		field.Name = f.Key
+		field.Labels = labels
+		field.Set(0, v)
+		s.fields = append(s.fields, field)
+	}
+	return nil
+}
+
+func tagsToLabels(tags []*influx.Tag) data.Labels {
+	labels := data.Labels{}
+	for i := 0; i < len(tags); i += 1 {
+		labels[tags[i].Key] = tags[i].Value
+	}
+	return labels
+}
+
+// append to existing metricFrame fields.
+func (s *metricFrame) append(m influx.Metric) error {
+	s.fields[0].Append(tagsToLabels(m.TagList()).String()) // TODO, use labels.String()
+	s.fields[1].Append(m.Time())
+
+	fields := m.FieldList()
+	sort.Slice(fields, func(i, j int) bool {
+		return fields[i].Key < fields[j].Key
+	})
+
+	for _, f := range fields {
+		ft, v, err := s.getFieldTypeAndValue(f)
+		if err != nil {
+			return err
+		}
+		if index, ok := s.fieldCache[f.Key]; ok {
+			field := s.fields[index]
+			if ft != field.Type() {
```

Comment:
This is the real fix -- the rest is just moving files from `grafana-live-sdk` to core